### ISSUE TITLE
Provider constraints are incompatible with the latest beta-private cluster submodule

### ISSUE THREAD

**Issue by clausstrommer** [terraform-google-modules/terraform-google-sql-db#536]
### TL;DR

The latest beta-private-cluster submodule has the version constraint `>= 5.0.0` of `hashicorp/google` and `hashicorp/google-beta`.  The SQL submodules have the version constraint `< 5.0.0` for those modules.

### Expected behavior

Google submodules should not have mutually exclusive provider version constraints.

### Observed behavior

Version constraint resolution fails with
```
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider
│ hashicorp/google: no available releases match the given constraints >=
│ 3.33.0, >= 3.39.0, >= 3.43.0, >= 3.53.0, >= 3.79.0, >= 3.83.0, >= 4.0.0, >=
│ 4.18.0, >= 4.25.0, >= 4.41.0, != 4.49.0, >= 4.51.0, >= 4.64.0, >= 4.80.0,
│ >= 5.0.0, < 5.0.0, < 6.0.0
╵

╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider
│ hashicorp/google-beta: no available releases match the given constraints >=
│ 3.43.0, >= 3.79.0, >= 4.64.0, >= 4.80.0, >= 5.0.0, < 5.0.0, < 6.0.0
```

### Terraform Configuration

```hcl
module "sql" {
  source  = "GoogleCloudPlatform/sql-db/google//modules/postgresql"
  version = ">= 15.0"

  (any minimal configuration you wish to use)
}

module "gke" {
  source  = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
  version = ">= 17.2"

  (any minimal configuration you wish to use)
}
```
```


### Terraform Version

```sh
1.5.x
```

### Additional information

Published dependency versions:  

* [GKE private-beta v29.0.0](https://registry.terraform.io/modules/terraform-google-modules/kubernetes-engine/google/latest/submodules/beta-private-cluster?tab=dependencies)
* [CloudSQL v17.0.0](https://registry.terraform.io/modules/GoogleCloudPlatform/sql-db/google/latest/submodules/postgresql?tab=dependencies)

**Comment by clausstrommer** [comment]
Additional info:  The version dependencies in `master` [have been updated](https://github.com/terraform-google-modules/terraform-google-sql-db/pull/523),  but this is not reflected in the published version of the submodules.

**Comment by navilg** [comment]
private-cluster module and postgresql module's google version's are conflicting. One needs hashicorp/google version >= 5.0.0 and one needs < 5.0.0. 

Any workaround we can use until next release ? I see its mitigated in master branch but not yet released.

**Comment by g-awmalik** [comment]
We'll be releasing upgraded modules support v5 soon. Please watch this [PR](https://github.com/terraform-google-modules/terraform-google-sql-db/pull/509)

**Comment by g-awmalik** [comment]
All modules should now be pulling tgp version >= 5 with v17.0.1
